### PR TITLE
Feature/crud question options

### PR DESCRIPTION
**Pull request by oonosan**

### CODE REVIEW

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/models/question-provisional.model.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { BaseModel } from './base.model';
```
Comment (guillemurano) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/models/question-provisional.model.ts:1]: Do we need this provisional model?

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/models/question.model.ts [hunk 2]
```diff
@@ -4,9 +4,10 @@ import { QuestionOption } from './question-option.modal';
 export class QuestionModel extends BaseModel {
   key: string;
   label: string;
-  required: boolean;
   order: number;
   controlType: string;
```
Comment (guillemurano) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/models/question.model.ts:8]: ControlType is linked to an entity, model must be added to reflect this

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/enum/controlTypes.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+export enum ControlType {
```
Comment (guillemurano) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/enum/controlTypes.ts:1]: This should be a model loaded from ControlType in the db

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/layout/admin-layout/admin-layout.module.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@ import { CommonModule } from '@angular/common';
 import { NgModule } from '@angular/core';
 import { IconDefinition } from '@ant-design/icons-angular';
 
-import { HeartOutline, ProfileOutline, QuestionOutline, TeamOutline } from '@ant-design/icons-angular/icons';
+import { HeartOutline, ProfileOutline, StarOutline, TeamOutline } from '@ant-design/icons-angular/icons';
```
Comment (guillemurano) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/layout/admin-layout/admin-layout.module.ts:5]: QuestionOutline shouldn't be removed

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/layout/admin-layout/admin-layout.module.ts [hunk 5]
```diff
@@ -23,7 +23,7 @@ import { HttpLoaderFactory } from '../../../../app.module';
 import { HttpClient } from '@angular/common/http';
 import { ReactiveFormsModule } from '@angular/forms';
 
-const ICONS: IconDefinition[] = [HeartOutline, ProfileOutline, QuestionOutline, TeamOutline];
```
Comment (guillemurano) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/layout/admin-layout/admin-layout.module.ts:26]: QuestionOutline shouldn't be removed

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.css [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+.priority-question-item {
+  margin-bottom: 1em;
+}
+
+.dynamic-delete-button {
+  cursor: pointer;
+  position: relative;
+  top: 4px;
```
Comment (guillemurano) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/priority-questions/question-create/questions-create.component.css:70]: favor using em

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/donations/donations-priority/donations-priority.component.ts [hunk 7]
```diff
@@ -1,19 +1,19 @@
...
+import { Question } from 'src/app/shared/models/question-provisional.model';
 @Component({
   selector: 'app-donations-priority',
   templateUrl: './donations-priority.component.html',
   styleUrls: ['./donations-priority.component.css'],
 })
 export class DonationPriorityComponent implements OnInit {
   form: FormGroup;
-  questions: QuestionModel[];
+  questions: Question[];
```
Comment (guillemurano) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/donations/donations-priority/donations-priority.component.ts:11]: Use QuestionModel instead

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/layout/admin-layout/admin-layout.component.ts [hunk 8]
```diff
@@ -14,7 +14,7 @@ export class AdminLayoutComponent implements OnInit, OnDestroy {
     { title: 'Admin.Menu.Title.Donation', url: './donations', iconType: 'heart' },
     { title: 'Admin.Menu.Title.User', url: './user', iconType: 'team' },
     { title: 'Admin.Menu.Title.Organization', url: './organization', iconType: 'profile' },
-    { title: 'Admin.Menu.Title.Question', url: './question', iconType: 'question' },
+    { title: 'Admin.Menu.Title.Priority-questions', url: './priority-questions', iconType: 'star' },
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/layout/admin-layout/admin-layout.component.ts:17]: I suggest to create an enumeration for iconType

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/models/question-provisional.model.ts (hunk 1 above)
Comment (guillemurano) [reply src/DonateTo.WebApplication/ClientApp/src/app/shared/models/question-provisional.model.ts:1]: @oonosan remove this file as it's no longer needed

**Pull request merged**